test/src/test/java/hudson/model/JobTest.java

```diff
@@ -286,4 +291,35 @@ private static void tryConfigDotXml(JenkinsRule.WebClient wc, int status, String
         p.renameTo("different-name");
     }
 
+    @Issue("JENKINS-30502")
+    @Test public void testRenameTrimsSpace() throws Exception {
+        tryRename("myJob", " foo", "foo");
```

Comment:
please use Parametrization instead of actually 2 cases in one method, or split it to separate methods, as of when fails first sentence, the second will never be run.